server/tests/purchase.test.js

```diff
@@ -0,0 +1,60 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const parallel = require('mocha.parallel');
+const app = require('../app');
+const mongoose = require('mongoose');
+
+chai.use(chaiHttp);
+let should = chai.should();
+let assert = chai.assert;
+
+
+describe('concurrency test on purchase', () => {
+
+    var ticketId = '';
```

Comment:
The mocha.parallel will send requests in one thread that could got succeed, but we should test on thread. 
Also, this challenges should be focusing on db translations rather than general operations like asyn/await.